### DISCUSSION TITLE
slugify a field

### DISCUSSION THREAD

**Discussion by calculusky** [prisma/prisma#10578] [Q&A] [answered]
please how do I generate a url slug from a field in prisma data model just like sequelize slugify with sequelize ORM
Thanks

**Comment by austincrim** [comment, marked as answer]
Hey @calculusky 👋,

Prisma does not have built-in support for doing operations like "slugify". You will have to use a separate operation to slugify your strings before saving them to your database.

You could use something like [this package](https://www.npmjs.com/package/slugify).

Let me know if you have questions!

**Reply by calculusky** [reply]
@austincrim thanks but how am i sure if the slug to be generated will not have a duplicate in the database resulting in violating unique field property

**Reply by austincrim** [reply]
You could run a `findUnique` query before you do any writes to check if that slug exists already.

**Comment by anis-marrouchi** [comment]
Thank you @austincrim for pointing out how we shall proceed for "slugify" operations.
I would also to share for anyone looking for a quick solution to use [middlewares](https://www.prisma.io/docs/concepts/components/prisma-client/middleware) 

Here is an example
```js
prisma.$use(async (params, next) => {
  if((params.action === 'create' || params.action === 'update') && ['Category', 'Product', 'YourModel'].includes(params.model)) {
    let {args:{data}} = params;
   // Check if slug exists by `findUnique` (did not test)
    data.slug = slugify(`${data.name}`, {lower: true, strict: true, remove: /[*+~.()'"!:@]/g});
  }
  const result = await next(params)
  return result
})
```
I've used slugify as suggested. Have it in your prisma.ts|js file. See [Best Practices](https://pris.ly/d/help/next-js-best-practices)

**Reply by justinhenricks** [reply]
This is great - the only caveat I'm finding is that ideally I want my slug field to be required, but if I leave the field required and run a graph QL mutation create without the slug argument, it appears the validation is running before this middleware and the mutation fails.

For now, I've made my slug fields optional and implemented the middleware, but doesn't seem ideal to not enforce not null on the slug column. Any suggestions?

**Comment by calculusky** [comment]
Thanks guys. I appreciate your contributions

**Comment by tiagobbraga** [comment]
+1

**Comment by bantawa04** [comment]
Now middleware is depreciated on newer version. We can use this


```
new PrismaClient().$extends({
    query: {
      $allModels: {
        create({ model, args, query }) {

          if (args.data && args.data?.title) {

            const slug = args.data?.title.toLowerCase().replace(/ /g, "-")

            args.data.slug = slug
          }
          return query(args)
        },
        update({ model, args, query }) {

          if (args.data && args.data?.title) {
            const slug = args.data?.title.toLowerCase().replace(/ /g, "-")
            args.data.slug = slug
          }

          return query(args)
        }
      }
    }
  })
  ```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
